Description: Compilation of static & shared lib.
 <long description that can span multiple lines, optional>
Author: <name and email of author, optional>
Origin: <upstream|backport|vendor|other>, <URL, required except if Author is present>
Bug: <URL to the upstream bug report if any, implies patch has been forwarded, optional>
Bug-<Vendor>: <URL to the vendor bug report if any, optional>
Forwarded: <URL|no|not-needed, useless if you have a Bug field, optional>
Applied-Upstream: <version|URL|commit, identifies patches merged upstream, optional>
Reviewed-by: <name and email of a reviewer, optional>
Last-Update: <YYYY-MM-DD, last update of the meta-information, optional>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/gatb-core/src/CMakeLists.txt
+++ b/gatb-core/src/CMakeLists.txt
@@ -79,16 +79,28 @@
 file (GLOB_RECURSE  ComponentFiles  *  ${PROJECT_BINARY_DIR}/src/gatb/*.cpp)
 
 # We add the compilation options for the library
-add_definitions (${gatb-core-flags})
-
+# PS: All required flags = ${gatb-core-flags} + ${CMAKE_CXX_FLAGS}
+# We don't want -flto flag on static lib so we add
+# specific flags to each target via COMPILE_FLAGS parameter.
+# Warning: This requires 2 independant compilations,
+# and this doubles the compilation time.
 include_directories (${gatb-core-includes})
 
-add_library    (gatbcore-static  STATIC  ${ComponentFiles} )
-#add_library   (gatbcore-dynamic SHARED  ${ComponentFiles} )
-
-target_link_libraries  (gatbcore-static hdf5 z pthread)
-set_target_properties  (gatbcore-static   PROPERTIES OUTPUT_NAME  gatbcore  clean_direct_output 1)
-#set_target_properties (gatbcore-dynamic  PROPERTIES OUTPUT_NAME  gatbcore  clean_direct_output 1)
+# Static lib
+# Remove -flto
+string(REPLACE "-flto" "" TEMP_CXX_FLAGS "${gatb-core-flags} ${CMAKE_CXX_FLAGS}")
+add_library   (gatbcore-static  STATIC   ${ComponentFiles})
+target_link_libraries (gatbcore-static   hdf5 z pthread)
+set_target_properties (gatbcore-static   PROPERTIES OUTPUT_NAME  gatbcore  clean_direct_output 1
+                                                    COMPILE_FLAGS ${TEMP_CXX_FLAGS})
+# Dynamic lib
+# Add -flto
+add_library   (gatbcore-dynamic SHARED   ${ComponentFiles})
+target_link_libraries (gatbcore-dynamic  hdf5 z pthread)
+set_target_properties (gatbcore-dynamic  PROPERTIES OUTPUT_NAME  gatbcore  clean_direct_output 1
+                                                    COMPILE_FLAGS "${TEMP_CXX_FLAGS} -flto"
+                                                    VERSION "${gatb-core-version}"
+                                                    SOVERSION "0")
 
 ################################################################################
 #  INSTALLATION 
@@ -96,7 +108,8 @@
 
 # We install the libraries
 IF (NOT DEFINED GATB_CORE_INSTALL_EXCLUDE)
-    install (TARGETS gatbcore-static DESTINATION lib)
+    install (TARGETS gatbcore-static  DESTINATION lib)
+    install (TARGETS gatbcore-dynamic DESTINATION lib)
     install (DIRECTORY ${PROJECT_SOURCE_DIR}/src/ DESTINATION include FILES_MATCHING PATTERN "*.hpp" PATTERN "*.tpp" PATTERN "*.pri" PATTERN "*.h")
     install (FILES ${PROJECT_BINARY_DIR}/include/gatb/system/api/config.hpp DESTINATION include/gatb/system/api/)
 ENDIF ()
